core/jazz_metrics/index.js

```diff
@@ -190,26 +194,47 @@ function validateAssets(assetsArray, eventBody) {
           }
         } else {
           var paramMetrics = [];
-          var getAssetNameDetails = utils.getNameSpaceAndMetricDimensons(assetItem.type);
+          var getAssetNameDetails = utils.getNameSpaceAndMetricDimensons(assetItem.type, assetItem.provider);
 
           if (!getAssetNameDetails.isError) {
             paramMetrics = getAssetNameDetails.paramMetrics;
-            exportable.getActualParam(paramMetrics, getAssetNameDetails.awsNameSpace, assetItem, eventBody)
+
+            if (assetItem.provider === 'aws') {
+              exportable.getActualParam(paramMetrics, getAssetNameDetails.nameSpace, assetItem, eventBody)
               .then(res => {
                 newAssetArray.push({
+                  "nameSpace": "aws",
                   "actualParam": res,
                   "userParam": assetItem
                 });
+                logger.debug("Validated Assets: " + JSON.stringify(newAssetArray));
                 resolve(newAssetArray);
               })
               .catch(error => {
+                logger.error(error);
                 reject(error);
               });
+            } else if (assetItem.provider === 'gcp') {
+              exportable.getApigeeParam(paramMetrics, eventBody)
+              .then(res => {
+                newAssetArray.push({
+                  "nameSpace": "gcp",
+                  "actualParam": res,
+                  "userParam": assetItem
+                });
+                logger.debug("Validated Assets: " + JSON.stringify(newAssetArray));
+                resolve(newAssetArray);
+              })
+              .catch(error => {
+                logger.error(error);
+                reject(error);
+              });
+            }
           } else {
             logger.error("Unsupported metric type. ");
             reject({
               result: "inputError",
-              message: "Unsupported metric type."
+              message: getAssetNameDetails.nameSpace
```

Comment:
if..else.. suggests that the error message could be `Unsupported asset provider: assetItem.provider`?

Reply:
not really. What you are seeing is the inner if/else

Reply:
updated